### DISCUSSION TITLE
Blog 3: Sequenz Diagram & Aktivitäts Diagram

### DISCUSSION THREAD

**Discussion by Max-Rohrhirsch** [DHBW-UMOC/UMOC#7] [Announcements]
# Blog 3
In diesem Blog gehen wir auf die Sequenzdiagramme, Aktivitätsdiagramme und Templates in Jira ein. 
Der Link zu unserer SRS ist hier: https://github.com/DHBW-UMOC/UMOC/edit/master/docs/UMOC_SRS.md 

### Sequenzdiagramm
Als Teil der Aufgabe haben wir ein Diagramm erstellt, welches zeigt, wie der Client, Server und die Datenbank miteinander kommunizieren am Beispiel von einem Login und Registrieren von neuen Nutzern. Die Kommunikation zwischen Server und Client geht über das Netzwerk (REST-API). Die Verknüpfung zwischen Server und Datenbanken geht über eine Library. Welche genau haben wir uns noch nicht entschieden.
![sequenzdiagram drawio](https://github.com/user-attachments/assets/a36b8467-5dd6-4100-9e44-6cc0be6214f4)

### Aktivitätsdiagramm
Anhand des Aktivitätsdiagramm sollen die 3 Prozesse gezeigt werden.
Der erste Prozess zeigt, wie das Einlogen im Hintergrund funktioniert.
Der zweite Prozess ist das Senden von Nachrichten.
Der Letzte Prozess zeigt, wie das ändern vom Passwort funktioniert.
![Aktivitäts_Diagramme drawio](https://github.com/user-attachments/assets/b6c4f15f-514c-46a7-b43d-4d24cc11b056)

### Jira Template
Damit die Storys alle ähnlich aussehen, haben wir ein Template erstellt, welches wir als Grundlage von jeder Story verwenden.
Das Template hat folgende Unterpunkte:
- Motivation: Warum ist die Story wichtig
- Lösungsvorschlag: Eine Idee, wie der Storyschreiber es sich vorstellt.
- Definition of Done: To-do-Liste, die bei fast jeder Story gebraucht werden (z. B. Readme aktualisieren, Unit-Tests Schreiben, …)
- To-do/Abnahmekriterien: Was der Entwickler machen muss, damit die Story fertig ist.

**Comment by bunki1224** [comment]
Mir gefallen die Grafiken sehr gut. Sie sind übersichtlich und eindeutig gestaltet, außerdem gefällt mir der Darkmode in den Grafiken wirklich gut. Der Ansatz templates für die Gira Tickets  ist wirklich sehr sinnvoll. Es beantwortet die wichtigsten Fragen und definiert klare Kriterien an die Story.

Etwas mehr Erläuterung zu den einzelnen Grafiken wäre toll gewesen. Trotzdem ein sehr gelungener Artikel.
>MFG das TripAhead Team (JB)
